spec/chewy/fields/base_spec.rb

```diff
@@ -400,6 +400,313 @@
+          ]
+
+          Country.create!(id: 1, cities: cities)
+        end
+
+        specify do
+          expect(CountriesIndex::Country.root.compose(country_with_cities)).to eq(
+            'id' => 1, 'cities' => [
+              {'id' => 1}
+            ]
+          )
+        end
+      end
+
+      context 'nested fields' do
+        context 'with ignore_blank: true flag' do
+          before do
+            stub_index(:countries) do
+              define_type Country do
+                field :id
+                field :cities, ignore_blank: true do
+                  field :id
+                  field :name
+                  field :historical_name, ignore_blank: true
+                  field :description
+                end
+              end
+            end
+          end
+
+          let(:country) { Country.create!(id: 1, cities: cities) }
+          context('without cities') do
+            let(:cities) { [] }
+            specify do
+              expect(CountriesIndex::Country.root.compose(country))
+                .to eq('id' => 1)
+            end
+          end
+          context('with cities') do
+            let(:cities) { [City.create!(id: 1, name: '', historical_name: '')] }
+            specify do
+              expect(CountriesIndex::Country.root.compose(country)).to eq(
+                'id' => 1, 'cities' => [
+                  {'id' => 1, 'name' => '', 'description' => nil}
+                ]
+              )
+            end
+          end
+        end
+
+        context 'with ignore_blank: false flag' do
+          before do
+            stub_index(:countries) do
+              define_type Country do
+                field :id
+                field :cities, ignore_blank: false do
+                  field :id
+                  field :name
+                  field :historical_name
+                  field :description
+                end
+              end
+            end
+          end
+
+          let(:country_with_cities) { Country.create!(id: 1) }
+
+          specify do
+            expect(CountriesIndex::Country.root.compose(country_with_cities))
+              .to eq('id' => 1, 'cities' => [])
+          end
+        end
+
+        context 'without ignore_blank: true flag' do
+          before do
+            stub_index(:countries) do
+              define_type Country do
+                field :id
+                field :cities do
+                  field :id
+                  field :name
+                  field :historical_name
+                  field :description
+                end
+              end
+            end
+          end
+
+          let(:country_with_cities) { Country.create!(id: 1) }
+
+          specify do
+            expect(CountriesIndex::Country.root.compose(country_with_cities))
+              .to eq('id' => 1, 'cities' => [])
+          end
+        end
+      end
+
+      # rubocop:disable Lint/SymbolConversion
+      context 'pass ignore_blank with string key' do
```

Comment:
Why should we test string keys?

Reply:
Just because firstly I added to #compose the check of 
`@options[:type].to_sym == :geo_point to allow users to pass types as string.` according to the next comment https://github.com/toptal/chewy/pull/778#discussion_r603470106

But after I removed this check since https://github.com/toptal/chewy/pull/778#discussion_r604161147

Do you think I can remove this specs now as unnecessary? @rabotyaga

Reply:
Yeah, I think it's redundant.